File: k-diff-pairs-in-an-array/k-diff-pairs-in-an-array.py
class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
        res = set()
        i = 0
        j = 1
        nums.sort()
        while j < len(nums):
            if nums[j]-nums[i] == k:
                res.add((nums[i], nums[j]))
                i += 1
                j = i+1
            elif nums[j]-nums[i] > k:
                i += 1
                j = i+1
            else:
                j += 1

        return len(res)

File: 532-k-diff-pairs-in-an-array/k-diff-pairs-in-an-array.cpp
class Solution {
public:
    int findPairs(vector<int>& nums, int k) {
        set<pair<int,int>>st ; 
        sort(nums.begin(),nums.end());
        int i = 0 ; 
        int j = 1;
        int ans = 0 ;
        while(j<nums.size()){
            int  diff = nums[j]-nums[i];
            if(diff==k){
                st.insert({nums[i],nums[j]});
                i++;
                j++;

            }
            else if(diff>k){
                i++;
                
            }
            
            else{
                j++;
            }
             if(i==j){
                j++;
            }

        }
        return st.size();

    }
};